### ISSUE TITLE
Convergence issue when nb_stacks > 1

### ISSUE THREAD

**Issue by AStupidBear** [philipperemy/keras-tcn#90]
If I change `nb_stacks` in `tasks/mnist_pixel/main.py` from 1 to 2, the accuracy will get stuck at around 10%. Is this an inherent problem of the network structure, or an implementation bug?

**Comment by philipperemy** [comment]
@psomers3 maybe linked to the recent changes.

**Comment by psomers3** [comment]
I don‘t seem to have this problem. I‘ve tested on Windows with:
Keras 2.3.1
Tensorflow (cpu) 1.14 and 2.0

**Comment by AStupidBear** [comment]
@psomers3 Could you post your progress result here?

I use a smaller dilation range ( `range(9)` -> `range(6)` but not `range(7)`) , the problem will disappear. I think it's an inherent problem of stacked TCNs, not an implementation bug.

**Comment by psomers3** [comment]
I can’t sign in to github from my computer, so unfortunately no copy-pasting. 
All I changed was nb_stacks=2 and after one epoch I result in

`loss: 4.7449 - accuracy: 0.8309 ...`

my model has 94,510 total trainable parameters. Does this match yours?

**Comment by AStupidBear** [comment]
Yes

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by psomers3** [comment]
But you still get a much lower accuracy? What version of Keras and tensorflow are you using? I really have no idea right now as to why we get different results.

**Comment by philipperemy** [comment]
2 stacks, current master
Trainable params: 94,510
```
60000/60000 [==============================] - 123s 2ms/step - loss: 8.7915 - accuracy: 0.8401 - val_loss: 0.1398 - val_accuracy: 0.9599
Epoch 2/100
60000/60000 [==============================] - 118s 2ms/step - loss: 0.1984 - accuracy: 0.9435 - val_loss: 0.1226 - val_accuracy: 0.9667
Epoch 3/100
60000/60000 [==============================] - 119s 2ms/step - loss: 0.1652 - accuracy: 0.9548 - val_loss: 0.1029 - val_accuracy: 0.9697
Epoch 4/100
60000/60000 [==============================] - 120s 2ms/step - loss: 0.1468 - accuracy: 0.9602 - val_loss: 0.0938 - val_accuracy: 0.9730
Epoch 5/100
60000/60000 [==============================] - 120s 2ms/step - loss: 0.1343 - accuracy: 0.9622 - val_loss: 0.0788 - val_accuracy: 0.9765
Epoch 6/100
60000/60000 [==============================] - 119s 2ms/step - loss: 0.1269 - accuracy: 0.9650 - val_loss: 0.0765 - val_accuracy: 0.9772
Epoch 7/100
60000/60000 [==============================] - 120s 2ms/step - loss: 0.1214 - accuracy: 0.9660 - val_loss: 0.0813 - val_accuracy: 0.9768
```

1 stack, current master

```
Trainable params: 47,170
60000/60000 [==============================] - 70s 1ms/step - loss: 0.4487 - accuracy: 0.8855 - val_loss: 0.1267 - val_accuracy: 0.9620
Epoch 2/100
60000/60000 [==============================] - 70s 1ms/step - loss: 0.1724 - accuracy: 0.9500 - val_loss: 0.0984 - val_accuracy: 0.9703
Epoch 3/100
60000/60000 [==============================] - 71s 1ms/step - loss: 0.1484 - accuracy: 0.9575 - val_loss: 0.1059 - val_accuracy: 0.9712
Epoch 4/100
60000/60000 [==============================] - 70s 1ms/step - loss: 0.1334 - accuracy: 0.9632 - val_loss: 0.0873 - val_accuracy: 0.9748
Epoch 5/100
60000/60000 [==============================] - 71s 1ms/step - loss: 0.1221 - accuracy: 0.9657 - val_loss: 0.0799 - val_accuracy: 0.9764
Epoch 6/100
60000/60000 [==============================] - 70s 1ms/step - loss: 0.1188 - accuracy: 0.9666 - val_loss: 0.0791 - val_accuracy: 0.9791
Epoch 7/100
60000/60000 [==============================] - 68s 1ms/step - loss: 0.1166 - accuracy: 0.9683 - val_loss: 0.0769 - val_accuracy: 0.9776
```

1 stack, previous master
```
Trainable params: 47,170
60000/60000 [==============================] - 64s 1ms/step - loss: 0.3676 - accuracy: 0.8915 - val_loss: 0.1213 - val_accuracy: 0.9631
Epoch 2/100
60000/60000 [==============================] - 57s 942us/step - loss: 0.1713 - accuracy: 0.9517 - val_loss: 0.0997 - val_accuracy: 0.9721
Epoch 3/100
60000/60000 [==============================] - 57s 950us/step - loss: 0.1447 - accuracy: 0.9587 - val_loss: 0.0880 - val_accuracy: 0.9747
Epoch 4/100
60000/60000 [==============================] - 57s 955us/step - loss: 0.1318 - accuracy: 0.9629 - val_loss: 0.0879 - val_accuracy: 0.9747
Epoch 5/100
60000/60000 [==============================] - 57s 956us/step - loss: 0.1234 - accuracy: 0.9653 - val_loss: 0.0761 - val_accuracy: 0.9754
Epoch 6/100
60000/60000 [==============================] - 57s 956us/step - loss: 0.1155 - accuracy: 0.9677 - val_loss: 0.0717 - val_accuracy: 0.9778
Epoch 7/100
60000/60000 [==============================] - 57s 956us/step - loss: 0.1158 - accuracy: 0.9671 - val_loss: 0.0641 - val_accuracy: 0.9810
```

2 stacks, previous master
```
60000/60000 [==============================] - 117s 2ms/step - loss: 0.4603 - accuracy: 0.8835 - val_loss: 0.1490 - val_accuracy: 0.9554
Epoch 2/100
60000/60000 [==============================] - 112s 2ms/step - loss: 0.1770 - accuracy: 0.9489 - val_loss: 0.1034 - val_accuracy: 0.9687
Epoch 3/100
60000/60000 [==============================] - 112s 2ms/step - loss: 0.1504 - accuracy: 0.9564 - val_loss: 0.1225 - val_accuracy: 0.9668
Epoch 4/100
60000/60000 [==============================] - 114s 2ms/step - loss: 0.1360 - accuracy: 0.9619 - val_loss: 0.0793 - val_accuracy: 0.9763
Epoch 5/100
60000/60000 [==============================] - 120s 2ms/step - loss: 0.1286 - accuracy: 0.9641 - val_loss: 0.1070 - val_accuracy: 0.9680
Epoch 6/100
60000/60000 [==============================] - 120s 2ms/step - loss: 0.1241 - accuracy: 0.9657 - val_loss: 0.0941 - val_accuracy: 0.9739
Epoch 7/100
60000/60000 [==============================] - 120s 2ms/step - loss: 0.1189 - accuracy: 0.9670 - val_loss: 0.0939 - val_accuracy: 0.9732
```

**Comment by philipperemy** [comment]
@AStupidBear sorry I could not reproduce the results on a GPU/CPU. I'll close this issue for now. Reinstall it, re-run and tell me if the issue still persists.
